libr/core/anal_tp.c

```diff
@@ -113,9 +113,12 @@ static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, char *type
 		//type *ptr => type *
 		strncat (ntype, " *", len);
 	}
-	if (ref && r_str_endswith (ntype, "*")) {
-		// char * => char **
-		strncat (ntype, "*", len);
+	if (ref) {
+		if (r_str_endswith (ntype, "*")) { // type * => type **
+			strncat (ntype, "*", len);
```

Comment:
dont use strncat,, and if u do it do it properly. because this can overflow in many ways. use RStrBuf instead